mongodump/mongodump_test.go

```diff
@@ -1024,3 +1047,110 @@ func TestMongoDumpOrderedQuery(t *testing.T) {
 		})
 	})
 }
+
+func TestMongoDumpViewsAsCollections(t *testing.T) {
+	testtype.SkipUnlessTestType(t, testtype.IntegrationTestType)
+	log.SetWriter(ioutil.Discard)
+
+	Convey("With a MongoDump instance", t, func() {
+		err := setUpMongoDumpTestData()
+		So(err, ShouldBeNil)
+
+		err = setUpDBView()
+		So(err, ShouldBeNil)
+
+		Convey("testing that the dumped directory contains information about metadata", func() {
+
+			md := simpleMongoDumpInstance()
+			md.ToolOptions.Namespace.DB = testDB
+			md.OutputOptions.Out = "dump"
+			md.OutputOptions.ViewsAsCollections = true
+
+			err = md.Init()
+			So(err, ShouldBeNil)
+
+			err = md.Dump()
+			So(err, ShouldBeNil)
+
+			path, err := os.Getwd()
+			So(err, ShouldBeNil)
+
+			dumpDir := util.ToUniversalPath(filepath.Join(path, "dump"))
+			dumpDBDir := util.ToUniversalPath(filepath.Join(dumpDir, testDB))
+			So(fileDirExists(dumpDir), ShouldBeTrue)
+			So(fileDirExists(dumpDBDir), ShouldBeTrue)
+
+			Convey("having one metadata file per read-only view", func() {
+				c1, err := countNonIndexBSONFiles(dumpDBDir)
+				So(err, ShouldBeNil)
+
+				c2, err := countMetaDataFiles(dumpDBDir)
+				So(err, ShouldBeNil)
+
+				So(c1, ShouldEqual, c2)
+
+			})
+
+			Reset(func() {
+				So(os.RemoveAll(dumpDir), ShouldBeNil)
+			})
+		})
+
+		Reset(func() {
+			So(tearDownMongoDumpTestData(), ShouldBeNil)
+		})
+
+	})
+}
+
+func TestMongoDumpViews(t *testing.T) {
```

Comment:
What's this test meant to be for?

Reply:
Ryan pointed out that if we dump a view without using the flag `--viewAsCollections`, we will attempt to run `listIndexes`, so I wrote this test to make sure views are being dumped correctly in the case that they are not read-only.

Do you think it's unnecessary/redundant?

Reply:
I see, I think there's a small bit of confusion here about the terminology of collections vs views. This test is dumping collections, not views. If `--viewsAsCollections` isn't set, then only collections are dumped. So the name of the test confused me a bit.

Just to clarify the terminology, there are two types of collection-like objects: regular collections which store data on disk, and views. Views are always read-only. They are a stored aggregation pipeline and they don't save any data to disk. 

(There are also _materialized views_, which do save the output of an aggregation to disk. Those can be read/write. But materialized views are just regular collections. The name describes the technique for populating the collection, not anything about the collection itself.)

I don't think this test is necessary since it's already covered by `TestMongoDumpMetaData()`.